### PR TITLE
[Neohub] Handle out of range values for temperatures

### PR DESCRIPTION
**Pull request by carlossg**
In some cases the neohub will return 255 and 127 values

Also handle unregistered neostats

### CODE REVIEW

File: bundles/binding/org.openhab.binding.neohub/src/main/java/org/openhab/binding/neohub/internal/NeoHubBinding.java [hunk 1]
```diff
@@ -110,10 +112,28 @@ protected void execute() {
...
+        if (deviceInfo == null) {
+            logger.warn("Failed to find device info for {}", deviceName);
+            return UnDefType.NULL;
+        }
         switch (property) {
             case CurrentTemperature:
-                return new DecimalType(deviceInfo.getCurrentTemperature());
+                BigDecimal currentTemperature = deviceInfo.getCurrentTemperature();
+                if (currentTemperature.intValue() == 255) {
+                    // when device is off or just reset we will get 255.255
```
Comment (9037568) [inline bundles/binding/org.openhab.binding.neohub/src/main/java/org/openhab/binding/neohub/internal/NeoHubBinding.java]: 255.255? Shouldn't this be just 255?

File: bundles/binding/org.openhab.binding.neohub/src/main/java/org/openhab/binding/neohub/internal/NeoHubBinding.java [hunk 2]
```diff
@@ -110,10 +112,28 @@ protected void execute() {
...
+                    logger.warn("Current temperature for device {} is out of range (is device off?): {}", deviceName,
```
Comment (9037568) [inline bundles/binding/org.openhab.binding.neohub/src/main/java/org/openhab/binding/neohub/internal/NeoHubBinding.java]: This should be a DEBUG, not a WARN.

File: bundles/binding/org.openhab.binding.neohub/src/main/java/org/openhab/binding/neohub/internal/NeoHubBinding.java [hunk 3]
```diff
@@ -110,10 +112,28 @@ protected void execute() {
...
+                            currentTemperature);
+                    return UnDefType.UNDEF;
+                }
+                return new DecimalType(currentTemperature);
             case CurrentFloorTemperature:
+                BigDecimal currentFloortemperature = deviceInfo.getCurrentFloorTemperature();
+                // when floor sensor is off we will get 127
+                if (currentFloortemperature.intValue() == 127) {
+                    logger.warn("Current floor temperature for device {} is out of range (is floor sensor off?): {}",
```
Comment (9037568) [inline bundles/binding/org.openhab.binding.neohub/src/main/java/org/openhab/binding/neohub/internal/NeoHubBinding.java]: This should be a DEBUG, not a WARN.

File: bundles/binding/org.openhab.binding.neohub/src/main/java/org/openhab/binding/neohub/internal/NeoHubBinding.java (hunk 1 above)
Comment (carlossg) [reply bundles/binding/org.openhab.binding.neohub/src/main/java/org/openhab/binding/neohub/internal/NeoHubBinding.java]: nope, we get a 255.255 value

**Comment by carlossg** [comment]
Sorry for the delay, addressed the comments

**Comment by 9037568** [comment]
Thanks, @carlossg !

**Pull request merged**